web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,426 @@
+/*
+   Copyright 2021 Snowplow Analytics Ltd. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+/*
+  MK_TRANSACTION
+  Side effects procedure.
+  Input: a concatenation of one or more DML sql statements split by semicolon.
+  It is important that the statements are not DDL.
+  Either all the statements in the block succeed and get committed or all get rolled back.
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.mk_transaction(VARCHAR);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.mk_transaction(DML_STATEMENTS VARCHAR)
```

Comment:
What's the value of having a procedure here instead of just wrapping the SQL in a `BEGIN .. END`  transaction?

Reply:
OK I found the docs you provide and looked into things like this: https://community.snowflake.com/s/question/0D50Z00008BDcR0SAL/how-do-you-commit-a-transaction-if-there-are-no-problems-but-roll-it-back-if-there-are

Makes sense now.